packages/expo-updates/src/UseUpdatesUtils.ts

```diff
@@ -0,0 +1,104 @@
+import * as Updates from './Updates';
+import type { UpdateEvent } from './Updates.types';
+import type { AvailableUpdateInfo, UpdatesInfo, UseUpdatesCallbacksType } from './UseUpdates.types';
+import { currentlyRunning } from './UseUpdatesConstants';
+
+/////// Internal functions ////////
+
+// Constructs the availableUpdate from the update manifest
+// Manifest is of type "any" until we no longer support classic updates
+export const availableUpdateFromManifest = (manifest: any) => {
+  return manifest
+    ? {
+        updateId: manifest?.id ?? null,
+        createdAt: manifest?.createdAt
+          ? new Date(manifest?.createdAt)
+          : manifest?.publishedTime
+          ? new Date(manifest?.publishedTime)
+          : null,
+        manifest,
+      }
+    : undefined;
+};
+
+// Constructs the UpdatesInfo from an event
+export const updatesInfoFromEvent = (
+  updatesInfo: UpdatesInfo | undefined,
+  event: UpdateEvent
+): UpdatesInfo => {
+  const lastCheckForUpdateTime = new Date();
+  if (event.type === Updates.UpdateEventType.NO_UPDATE_AVAILABLE) {
+    return {
+      ...updatesInfo,
+      currentlyRunning,
+      availableUpdate: undefined,
+      error: undefined,
+      lastCheckForUpdateTimeSinceRestart: lastCheckForUpdateTime,
+    };
+  } else if (event.type === Updates.UpdateEventType.UPDATE_AVAILABLE) {
+    return {
+      ...updatesInfo,
+      currentlyRunning,
+      availableUpdate: availableUpdateFromManifest(event.manifest),
+      error: undefined,
+      lastCheckForUpdateTimeSinceRestart: lastCheckForUpdateTime,
+    };
+  } else {
+    // event type === ERROR
+    return {
+      ...updatesInfo,
+      currentlyRunning,
+      availableUpdate: undefined,
+      error: new Error(event.message),
+      lastCheckForUpdateTimeSinceRestart: lastCheckForUpdateTime,
+    };
+  }
+};
+
+// Implementation of checkForUpdate
+export const checkForUpdateAndReturnAvailableAsync: (
+  callbacks?: UseUpdatesCallbacksType
+) => Promise<AvailableUpdateInfo | undefined> = async (callbacks) => {
+  try {
+    callbacks?.onCheckForUpdateStart && callbacks?.onCheckForUpdateStart();
+    const checkResult = await Updates.checkForUpdateAsync();
```

Comment:
We may want to rename `UpdateCheckResultFailure` to `UpdateCheckResultNoUpdateAvailable` or something like that. At first I was confused about why we didn't handle it as a failure here but in reality it isn't a failure per se.

Reply:
True, but I do worry about changing an API that has been the same for awhile